### DISCUSSION TITLE
Eslint Custom Rule Gets Detected In a Different File

### DISCUSSION THREAD

**Discussion by turturmurmur** [eslint/eslint#16873] [Q&A]
### Environment

Node version: v14.18.1
npm version: v6.14.15
Local ESLint version: v8.33.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

@typescript-eslint/parser

### What did you do?

Created eslint, prettier, typescript. Configured it with eslintrc.json file. 
I can run the command `npx eslint --ext .json jsonFolder/a.json` and I can see the error in the command line even with the line numbers it returns, but when it comes to seeing it inside the vscode I get an error like this below.
![image](https://user-images.githubusercontent.com/89113207/217122315-be7ec9a0-dfca-4db1-92de-28eaf864c6bd.png)
![image](https://user-images.githubusercontent.com/89113207/217122641-a7c37273-080a-4db9-b50b-eced1ef0b7a8.png)

I also see the errors on the problems panel in vscode but even that is showing to the wrong file.
![image](https://user-images.githubusercontent.com/89113207/217123200-260da3e0-0a98-445b-a264-fce0ffdfc1cc.png)

Sending repo link of the project for full reproducability below.

[Here's the code in a github repo, please use the branch customRuleWorksWithLineNumber
](https://github.com/turturmurmur/prettier-eslint-typescript-setup/tree/customRuleWorksWithLineNumber)
I have eslint and prettier plugins installed for vscode. 
### What did you expect to happen?

I was expecting the rule error to be found in a.json and b.json files

### What actually happened?

The rules found in the file1.ts file

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

When I remove the code and even put a false error with no logic, I see the same error in file1.ts file instead it should be observed in a.json and b.json files.

**Comment by Rec0iL99** [comment]
Hi @turturmurmur, I would suggest you to host your project files on GitHub or CodeSandbox for team members to review rather than having to download the zip file to view the code. Thanks.

**Comment by turturmurmur** [comment]
Hi @Rec0iL99 , I have edited my comment above and added the repo link as well. Thank you for the suggestion.

**Comment by aladdin-add** [comment]
eslint/ts-eslint was designed to lint valid js/ts, you cannot use it to lint json(as it's not valid js).

**Comment by turturmurmur** [comment]
@aladdin-add , in that case how come there are multiple plugins that work with eslint such as [eslint-plugin-json](https://www.npmjs.com/package/eslint-plugin-json) ?
because even eslint allows to lint with --ext .json which can be seen in this official site [link](https://eslint.org/docs/latest/use/configure/configuration-files#specifying-target-files-to-lint).

**Comment by aladdin-add** [comment]
oops, sorry! I missed you had wrote a custom plugin to lint json.

imo, the rule should be only applied to json, not js/ts. you can use `overrides`:

```js
{
  "overrides": [{
     "files": ["**/*.json"],
     "rules": {"no-duplicate-values/no-duplicate-values": "error"}
  }]
}
```

another issue is you were using `@typescript-eslint/parser` to parse json - which is not expected.
maybe you can check how `eslint-plugin-json` works. I'm guessing (but not sure) it's using a custom parser to generate estree-like ast.

**Comment by turturmurmur** [comment]
Good idea @aladdin-add , 
I have tried that logic and still getting the same problem 
![image](https://user-images.githubusercontent.com/89113207/217401664-09dbcccb-86aa-4ec1-b74c-5044ffb6f507.png)
and if I take out the rules then the rule disappeared 
![image](https://user-images.githubusercontent.com/89113207/217401727-e619832d-16a5-4b56-946b-3b80cca3ea86.png)

So none worked. I will also try how other plugins are doing it.
I am open for new ideas from anyone.

**Comment by Rec0iL99** [comment]
> Created eslint, prettier, typescript. Configured it with eslintrc.json file. I can run the command `npx eslint --ext .json jsonFolder/a.json` and I can see the error in the command line even with the line numbers it returns, but when it comes to seeing it inside the vscode I get an error like this below.

Hey @turturmurmur, since you mentioned that `eslint-cli` works as intended, then I guess you can consult the eslint -VSCode extension maintainers who will have a better idea as to how to help you here.

**Comment by turturmurmur** [comment]
Hi @Rec0iL99 , there is an issue on the CLI as well, because when I run it on a single file, it finds the issues on other files as well. So it is working out of scope. I believe it is related to .eslintrc.json file configuration but couldn't figure out what is happening.

**Comment by aladdin-add** [comment]
> there is an issue on the CLI as well, because when I run it on a single file, it finds the issues on other files as well. 

the issue here is you had manually check all the json files: https://github.com/turturmurmur/prettier-eslint-typescript-setup/blob/2b30692c330ed72544f04ecff2c31d8d548d07a1/plugins/eslint-plugin-no-duplicate-values.js#L20-L21

basically you should only check the passed json, not all. (you can get the filename using `context.getFilename()`)

**Reply by turturmurmur** [reply]
If I change that to only what the file name is being passed, then I will lose the option to check all json file checks with vscode eslint plugin

**Reply by aladdin-add** [reply]
1. you can use eslint's `overrdies` to specify which files should be linted.
2. vscode-eslint will respect eslint configs. If I recall correctly, vscode-eslint only lint js/ts by default, you still need to config something to work with json.
